include/dlaf/matrix/copy.h

```diff
@@ -37,8 +36,8 @@ void copy(MatrixTypeSrc<const T, device>& source, MatrixTypeDst<T, device>& dest
 
   for (SizeType j = 0; j < local_tile_cols; ++j)
     for (SizeType i = 0; i < local_tile_rows; ++i)
-      hpx::dataflow(hpx::util::unwrapping(dlaf::copy<T>), source.read(LocalTileIndex(i, j)),
-                    dest(LocalTileIndex(i, j)));
+      hpx::dataflow([](auto&& source,
+                       auto&& destination) { dlaf::copy<>(source.get(), destination.get()); },
```

Comment:
This wrapping lambda is only here because of overload resolution not working through `unwrapping`. An alternative would be to rename the `Tile` overload (or the `Matrix` overload) to e.g. `copy_tile`.